### PR TITLE
Remove the old createBlock API

### PR DESCRIPTION
**Pull request by yyongycy**
Remove redundant createBlock API

### CODE REVIEW

**Comment by yyongycy** [comment]
Removing the old createBlock API

**Comment by yyongycy** [comment]
@apc999  this is to remove the old createBlock API.

**Comment by yyongycy** [comment]
Most likely it will cause more conflict in OS->EE sync, better to merge it after 2.8 release cut.

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java [hunk 1]
```diff
@@ -69,8 +70,8 @@ public static BlockWorkerDataWriter create(final FileSystemContext context, long
     Preconditions.checkNotNull(blockWorker, "blockWorker");
     long sessionId = SessionIdUtils.createSessionId();
     try {
-      blockWorker.createBlock(sessionId, blockId, options.getWriteTier(), options.getMediumType(),
-          reservedBytes);
+      blockWorker.createBlock(sessionId, blockId, options.getWriteTier(),
+          new CreateBlockOptions("", options.getMediumType(), reservedBytes));
```
Comment (LuQQiu) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java:74]: Can you help add  @Nullable to the alluxiopath and medium input?
```
  public CreateBlockOptions(String alluxioPath, String medium, long initialBytes) {
```
null is the default or "" is the default?

**Review by LuQQiu — COMMENTED** [review]
can we have a consolidated default value when alluxioPath, medium is not set?
Currently, i see CreateBlockOptions use null but our tests use empty string "", better to only have one way

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java (hunk 1 above)
Comment (yyongycy) [reply core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java:74]: added

**Comment by yyongycy** [comment]
alluxio-bot, merge this please!

**Pull request merged**